lib/jxl/dec_external_image.cc

```diff
@@ -349,17 +347,12 @@ Status ConvertToExternal(const jxl::ImageBundle& ib, size_t bits_per_sample,
           },
           [&](const int task, int thread) {
             const int64_t y = task;
-            const float* JXL_RESTRICT row_in[4];
-            size_t c = 0;
-            for (; c < color_channels; c++) {
-              row_in[c] = color->PlaneRow(c, y);
-            }
-            if (want_alpha) {
-              row_in[c++] = ib.HasAlpha() ? alpha->Row(y) : ones.Row(0);
+            const float* JXL_RESTRICT row_in[kConvertMaxChannels];
+            for (size_t c = 0; c < num_channels; c++) {
+              row_in[c] = channels[c] ? channels[c]->Row(y) : ones.Row(0);
             }
-            JXL_ASSERT(c == num_channels);
-            hwy::float16_t* JXL_RESTRICT row_f16[4];
-            for (size_t r = 0; r < c; r++) {
+            hwy::float16_t* JXL_RESTRICT row_f16[kConvertMaxChannels];
+            for (size_t r = 0; r < num_channels; r++) {
```

Comment:
small nit (and fine to keep to prevent touching more code), now one could as well use 'c' as the loop variable everywhere for consistency with the other loops

Reply:
Yeah, that's a good point, I'll change it.